vault/resource_identity_entity.go

```diff
@@ -0,0 +1,198 @@
+package vault
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/vault/api"
+	"github.com/terraform-providers/terraform-provider-vault/util"
+)
+
+const identityEntityPath = "/identity/entity"
+
+func identityEntityResource() *schema.Resource {
+	return &schema.Resource{
+		Create: identityEntityCreate,
+		Update: identityEntityUpdate,
+		Read:   identityEntityRead,
+		Delete: identityEntityDelete,
+		Exists: identityEntityExists,
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "Name of the entity.",
+				ForceNew:    true,
+			},
+
+			"metadata": {
+				Type:        schema.TypeMap,
+				Optional:    true,
+				Description: "Metadata to be associated with the entity.",
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+
+			"policies": {
+				Type:     schema.TypeList,
+				Optional: true,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+				Description: "Policies to be tied to the entity.",
+			},
+
+			"disabled": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "Whether the entity is disabled. Disabled entities' associated tokens cannot be used, but are not revoked.",
+			},
+
+			"id": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "ID of the entity.",
+			},
+		},
+	}
+}
+
+func identityEntityUpdateFields(d *schema.ResourceData, data map[string]interface{}) {
+	if policies, ok := d.GetOk("policies"); ok {
+		data["policies"] = policies
+	}
+
+	if metadata, ok := d.GetOk("metadata"); ok {
+		data["metadata"] = metadata
+	}
+
+	if disabled, ok := d.GetOk("disabled"); ok {
+		data["disabled"] = disabled
+	}
```

Comment:
Not a blocker at all, but if you feel like it, updating an entity's `name` is also supported.